### PR TITLE
Update Scorecard Monitor workflow + fix Scorecard Report links

### PR DESCRIPTION
**Pull request by lelia**
- Updates Scorecard Monitor workflow action dependencies
- Updates location of Scorecard Monitor action to OSSF ✨
- Fixes Scorecard Report links to use new API visualizer URL

### CODE REVIEW

File: reports/scorecard/report.md [hunk 1]
```diff
@@ -4,79 +4,79 @@
...
+| [cisco-ospo/oss-template-go](https://github.com/cisco-ospo/oss-template-go) | [fbb045c](https://github.com/cisco-ospo/oss-template-go/commit/fbb045cbab4920e93f8bb90ac0abfe4cdaa156bd) | 4.9 | 2024-05-11T01:45:57Z | 0 / [Details](https://ossf.github.io/scorecard-visualizer/#/projects/github.com/cisco-ospo/oss-template-go/compare/fbb045cbab4920e93f8bb90ac0abfe4cdaa156bd/fbb045cbab4920e93f8bb90ac0abfe4cdaa156bd) | [View](https://ossf.github.io/scorecard-visualizer/#/projects/github.com/cisco-ospo/oss-template-go/commit/fbb045cbab4920e93f8bb90ac0abfe4cdaa156bd) | [Fix it](https://app.stepsecurity.io/securerepo?repo=cisco-ospo/oss-template-go) |
 
-_Report generated by [UlisesGascon/openssf-scorecard-monitor](https://github.com/UlisesGascon/openssf-scorecard-monitor)._
\ No newline at end of file
+_Report generated by [UlisesGascon/openssf-scorecard-monitor](https://github.com/UlisesGascon/openssf-scorecard-monitor)._
```
Comment (justaugustus) [inline reports/scorecard/report.md:82]:
```suggestion
_Report generated by [OpenSSF Scorecard Monitor](https://github.com/ossf/scorecard-monitor)._
```

**Review by justaugustus — CHANGES_REQUESTED** [review]
@lelia — One nit. Can you fix this upstream as well?

File: reports/scorecard/report.md (hunk 1 above)
Comment (lelia) [reply reports/scorecard/report.md:82]: Good catch! Can do 🫡

**Review by justaugustus — APPROVED** [review]
Thanks for the fixes, @lelia!

File: reports/scorecard/report.md (hunk 1 above)
Comment (lelia) [reply reports/scorecard/report.md]: Btw, looks like this was _partially_ [fixed upstream](https://github.com/ossf/scorecard-monitor/pull/75) already, but there was a find/replace error on the `issue.ejs` template: https://github.com/ossf/scorecard-monitor/pull/75/files#diff-32816260c1e1ed95d3ecb3128ec495d71816e45440ae2d0a189c139f762e91f8. 

Submitted upstream fix as https://github.com/ossf/scorecard-monitor/pull/83.

**Pull request merged**